website_form_recaptcha/models/website_form_recaptcha.py

```diff
@@ -57,12 +57,10 @@ def action_validate(self, response, remote_ip):
 
         for error in res.get('error-codes', []):
             raise ValidationError(
-                self.ERROR_MAP.get(
-                    error, self.ERROR_MAP[None]
-                )
+                self._get_error_message(error)
```

Comment:
raising inside a loop? Weird...

From a UX perspective, changing to something like this would mean a big boost:

```py
msg = "\n".join(self._get_error_message(error) 
                        for error in res.get('error-codes', []))
if msg:
    raise ValidationError(msg)
```

Reply:
For me that's the limit between refactoring and migration; you're asking me to change the implementation...

I agree with you, of course; I'd rather have all error messages displayed at once.

Can be done ithen.

Reply:
I guess migrations sometimes allow us to see problems that passed unnoticed before :blush:

Reply:
hey @dbo-odoo any news?

Reply:
@Yajo I don't think that @dbo-odoo will refactor the requested changes.
Can you accept the PR and I will do a new one with the refactoring?

Reply:
Red status is due to this PR? Maybe rebasing changes the thing.

Reply:
It's this PR, yes.

Reply:
OK then, anyone can fix it?

Reply:
@Yajo You are right. I didn't notice the red flag on Travis

Reply:
@pedrobaeza I can do the job, but @dbo-odoo will need to accept my PR into is fork in order to update this PR or I'm wrong?

Reply:
In this case, you can add a new PR directly to OCA that superseeds this, and we close this one.

Reply:
@pedrobaeza new PR https://github.com/OCA/website/pull/458